core/Utils.vala

```diff
@@ -699,8 +699,33 @@ namespace Maya.Util {
      * ical Exportation
      */
 
-    public void save_temp_selected_calendars (){
-        //TODO:Create the code !
+    public void save_temp_selected_calendars () {
+        var calmodel = Model.CalendarModel.get_default ();
+        var events = calmodel.get_events ();
+        var builder = new StringBuilder ();
+        builder.append ("BEGIN:VCALENDAR\n");
+        builder.append ("VERSION:2.0\n");
+        foreach (E.CalComponent event in events) {
+            builder.append (event.get_as_string ());
+        }
+        builder.append ("END:VCALENDAR");
+        string output = builder.str;   
```

Comment:
This is not required; you can get the data directory from the StringBuilder with uint8[] data =  builder.data;

Reply:
Thanks, I have now switched to using this.